app/broadcast_message/rest.py

```diff
@@ -155,10 +155,48 @@ def update_broadcast_message_status(service_id, broadcast_message_id):
     _update_broadcast_message(broadcast_message, new_status, updating_user)
     dao_update_broadcast_message(broadcast_message)
 
-    if new_status == BroadcastStatusType.BROADCASTING:
-        send_broadcast_message.apply_async(
-            kwargs={'broadcast_message_id': str(broadcast_message.id)},
-            queue=QueueNames.NOTIFY
-        )
+    if new_status in {BroadcastStatusType.BROADCASTING, BroadcastStatusType.CANCELLED}:
+        _create_broadcast_event(broadcast_message)
 
     return jsonify(broadcast_message.serialize()), 200
+
+
+def _create_broadcast_event(broadcast_message):
+    """
+    Creates a broadcast event, stores it in the database, and triggers the task to send the CAP XML off
+    """
+    msg_types = {
+        BroadcastStatusType.BROADCASTING: BroadcastEventMessageType.ALERT,
+        BroadcastStatusType.CANCELLED: BroadcastEventMessageType.CANCEL,
+    }
+
+    if broadcast_message.status == BroadcastStatusType.CANCELLED:
+        transmitted_finishes_at = broadcast_message.cancelled_at
+    else:
+        transmitted_finishes_at = broadcast_message.finishes_at
+
+    # TODO: This doesn't support placeholders yet. We shouldn't use BroadcastMessageTemplate when we add placeholders
+    # as that just outputs XML, we need the raw text.
+    event = BroadcastEvent(
+        service=broadcast_message.service,
+        broadcast_message=broadcast_message,
+        message_type=msg_types[broadcast_message.status],
+        transmitted_content={"body": broadcast_message.template.content},
+        transmitted_areas=broadcast_message.areas,
+        # TODO: Probably move this somewhere more standalone too and imply that it shouldn't change. Should it include
+        # a service based identifier too? eg "flood-warnings@notifications.service.gov.uk" or similar
```

Comment:
so if it shouldn't change, we should probs store it in config?

But I like the idea of using from address for service as an identifier.

Reply:
agreed - didnt see point of putting it in config now. i dont know if it's expected that it's a real email address. for example the US weather warnings have the sender `w-nws.webmaster@noaa.gov` (https://alerts-v2.weather.gov/#/). Again, just wanted a placeholder to get things mergeable ahead of finding out stuff, especially around is this value viewable to end users.

Reply:
makes sense 👍